tests/compliance_operator_v2_test.go

```diff
@@ -88,6 +152,148 @@ func waitForComplianceSuiteToComplete(t *testing.T, suiteName string, interval,
+	client := createDynamicClient(t)
+	scanSetting := &complianceoperatorv1.ScanSetting{}
+	scanSettingBinding := &complianceoperatorv1.ScanSettingBinding{}
+	err := client.Get(ctx, types.NamespacedName{Name: resourceName, Namespace: namespace}, scanSetting)
+	if err == nil {
+		_ = client.Delete(ctx, scanSetting)
+	}
+	err = client.Get(ctx, types.NamespacedName{Name: resourceName, Namespace: namespace}, scanSettingBinding)
+	if err == nil {
+		_ = client.Delete(ctx, scanSettingBinding)
+	}
+}
+
+func assertScanSetting(t *testing.T, scanConfig v2.ComplianceScanConfiguration, scanSetting *complianceoperatorv1.ScanSetting) {
+	require.NotNil(t, scanSetting)
+	cron, err := schedule.ConvertToCronTab(service.ConvertV2ScheduleToProto(scanConfig.GetScanConfig().GetScanSchedule()))
+	require.NoError(t, err)
+	assert.Equal(t, scanConfig.GetScanName(), scanSetting.GetName())
+	assert.Equal(t, cron, scanSetting.ComplianceSuiteSettings.Schedule)
+}
+
+func assertScanSettingBinding(t *testing.T, scanConfig v2.ComplianceScanConfiguration, scanSettingBinding *complianceoperatorv1.ScanSettingBinding) {
+	require.NotNil(t, scanSettingBinding)
+	assert.Equal(t, scanConfig.GetScanName(), scanSettingBinding.GetName())
+	for _, profile := range scanSettingBinding.Profiles {
+		assert.Contains(t, scanConfig.GetScanConfig().GetProfiles(), profile.Name)
+	}
+}
+
+func TestComplianceV2CentralSendsScanConfiguration(t *testing.T) {
+	ctx := context.Background()
+	k8sClient := createK8sClient(t)
+
+	conn := centralgrpc.GRPCConnectionToCentral(t)
+	// Create the ScanConfiguration service
+	service := v2.NewComplianceScanConfigurationServiceClient(conn)
+
+	// Get cluster ID
+	serviceCluster := v1.NewClustersServiceClient(conn)
+	clusters, err := serviceCluster.GetClusters(ctx, &v1.GetClustersRequest{})
+	assert.NoError(t, err)
+	require.Greater(t, len(clusters.GetClusters()), 0)
+	clusterID := clusters.GetClusters()[0].GetId()
+
+	// Set the cluster ID
+	scanConfig.Clusters = []string{clusterID}
+
+	// Scale down Sensor
+	assert.NoError(t, scaleToN(ctx, k8sClient, "sensor", stackroxNamespace, 0))
+
+	// Create ScanConfig in Central
+	res, err := service.CreateComplianceScanConfiguration(ctx, &scanConfig)
+	assert.NoError(t, err)
+
+	// Cleanup just in case the test fails
+	t.Cleanup(func() {
+		reqDelete := &v2.ResourceByID{
+			Id: res.GetId(),
+		}
+		_, _ = service.DeleteComplianceScanConfiguration(ctx, reqDelete)
+		cleanUpResourcesAndWait(ctx, t, scanName, coNamespace)
+	})
+
+	// Scale up Sensor
+	assert.NoError(t, scaleToN(ctx, k8sClient, "sensor", stackroxNamespace, 1))
+
+	// Assert the ScanSetting and the ScanSettingBinding are created
+	scanSetting := &complianceoperatorv1.ScanSetting{}
+	scanSettingBinding := &complianceoperatorv1.ScanSettingBinding{}
+	assertResourceDoesExist(ctx, t, scanName, coNamespace, scanSetting)
+	assertResourceDoesExist(ctx, t, scanName, coNamespace, scanSettingBinding)
```

Comment:
Looking at the CI output these assertions are never fulfilled. I see `scanSetting` and `scanSettingBinding` being used only within assertions, do they perhaps need to be applied or sent first?

Reply:
ScanSettings and ScanSettingBindings are created by sensor, which is what we want to test here. Applying them directly in the test would defeat the purpose of the test.